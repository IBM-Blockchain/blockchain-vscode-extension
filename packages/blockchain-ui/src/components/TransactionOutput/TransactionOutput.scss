@import '@carbon/type/scss/font-face/_mono';
@import '@carbon/layout/scss/_spacing';

.output-panel {
    position: fixed;
    height: 100%;
    width: 52%;
    background-color: #F4F4F4;
    padding: $spacing-07;

    .output-title {
        font-size: 0.875rem;
        color: #393939;
    }

    .output-body {
        @include carbon--font-face-mono;
        font-family: 'IBM Plex Mono';
        font-size: 0.875rem;
        color: inherit;
    }
}
