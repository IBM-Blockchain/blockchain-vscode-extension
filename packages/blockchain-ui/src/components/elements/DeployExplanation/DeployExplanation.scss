@import '@carbon/colors/scss/colors';
@import '@carbon/layout/scss/_spacing';

#root {

    .step-title {
        font-size: .875rem;
    }
    
    .title-and-tag {
        display: flex;
        align-items: center;

        span {
            margin-right: $spacing-05;
            font-size: .875rem;
        }

        .fabric-2-tag {
            background-color: $blue-60;
            color: #FFFFFF
        }
    }

    @media (max-width: 594px) {
        .title-and-tag {
            display: flex;
            flex-direction: column;
            align-items: flex-start;

            span {
                margin-bottom: $spacing-05;
            }
        }
    }

}