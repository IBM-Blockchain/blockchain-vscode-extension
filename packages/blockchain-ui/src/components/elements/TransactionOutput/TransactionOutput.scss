@import '@carbon/type/scss/font-face/_mono';
@import '@carbon/layout/scss/_spacing';

.output-panel {
    position: fixed;
    top: 0px;
    height: 100%;
    width: 55%;
    background-color: #F4F4F4;

    .output-panel-inner {
        margin: $spacing-07;
        padding-right: $spacing-07 + $spacing-09;


        .output-title {
            font-size: 0.875rem;
            color: #393939;
        }

        .output-body {
            @include carbon--font-face-mono;
            font-family: 'IBM Plex Mono', 'IBM Plex Sans', 'IBMPlexSans', Arial, Helvetica, sans-serif;
            font-size: 0.875rem;
            color: inherit;
        }

    }
}
