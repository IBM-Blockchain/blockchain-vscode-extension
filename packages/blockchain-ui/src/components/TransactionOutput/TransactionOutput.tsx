import React, { Component } from 'react';
import './TransactionOutput.scss';

interface OutputProps {
    output: string;
}

interface OutputState {
    output: string;
}

class TransactionOutput extends Component<OutputProps, OutputState> {
    constructor(props: Readonly<OutputProps>) {
        super(props);
        this.state = {
            output: this.props.output
        };
        this.determineOutput = this.determineOutput.bind(this);
    }

    componentDidUpdate(prevProps: OutputProps): void {
        if (prevProps.output !== this.props.output) {
            this.setState({
                output: this.props.output
            });
        }
    }

    determineOutput(): Array<JSX.Element>  {
        const outputArray: Array<JSX.Element> = [];
        if (this.state.output === '') {
            outputArray.push(<p className='output-body'>No transaction output available. Submit/evaluate to produce an output.</p>);
        } else {
            const outputStrings: Array<string> = this.state.output.split('\n');
            for (const output of outputStrings) {
                outputArray.push(<p className='output-body'>{output}</p>);
            }
        }
        return outputArray;
    }

    render(): JSX.Element {
        return (
            <div className='output-panel' id='output-panel'>
                <p className='output-title'>Output for untitled transaction</p>
                {this.determineOutput()}
            </div>
        );
    }
}

export default TransactionOutput;
