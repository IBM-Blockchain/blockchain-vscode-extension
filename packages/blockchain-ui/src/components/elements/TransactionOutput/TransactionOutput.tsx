import React, { FunctionComponent } from 'react';
import { Launch32 } from '@carbon/icons-react';
import { ExtensionCommands } from '../../../ExtensionCommands';
import CommandLink from '../CommandLink/CommandLink';
import './TransactionOutput.scss';
import block from '../../../resources/block.svg';

interface IProps {
    output: string;
}

const TransactionOutput: FunctionComponent<IProps> = ({ output }) => {
    return (
        <div className='output-panel' id='output-panel'>
            <p className='output-title'>Transaction output</p>
            <div className='output-panel-inner'>
                {output
                    ?  output.split('\n').map((line) => <p className='output-body'>{line}</p>)
                    : (
                        <div className='output-placeholder-container'>
                            <div className='output-placeholder'>
                                <img src={block} alt='block icon' />
                                <h4>No transaction output, yet!</h4>
                                <p>Submit or evaluate a transaction to view it's output here.</p>
                                <CommandLink
                                    linkContents={<>Learn more <Launch32/></>}
                                    commandName={ExtensionCommands.OPEN_TUTORIAL_PAGE}
                                    commandData={['Basic tutorials', 'A4: Invoking a smart contract from VS Code']}
                                />
                            </div>
                        </div>
                    )
                }
            </div>
        </div>
    );
};

export default TransactionOutput;
