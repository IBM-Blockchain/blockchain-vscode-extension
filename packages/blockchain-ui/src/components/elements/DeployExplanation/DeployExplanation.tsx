import React, { FunctionComponent } from 'react';
import { Accordion, AccordionItem, Link, Tag } from 'carbon-components-react';
import ThemedImage from '../ThemedImage/ThemedImage';
import lightStep1 from '../../../resources/lightStep1.svg';
import lightStep2 from '../../../resources/lightStep2.svg';
import lightStep3 from '../../../resources/lightStep3.svg';
import lightStep4 from '../../../resources/lightStep4.svg';
import darkStep1 from '../../../resources/darkStep1.svg';
import darkStep2 from '../../../resources/darkStep2.svg';
import darkStep3 from '../../../resources/darkStep3.svg';
import darkStep4 from '../../../resources/darkStep4.svg';
import './DeployExplanation.scss';

interface IProps  {
    showV2Explanation: boolean;
}

const DeployExplanation: FunctionComponent<IProps>  = ({showV2Explanation}) => {
    let explanationJSX: JSX.Element = <></>;

    if (showV2Explanation) {
        explanationJSX = (
            <Accordion>
                <AccordionItem title={'How does Fabric v2.X smart contract deployment work?'}>
                    <div className='bx--row margin-top-05 margin-bottom-09'>
                        <p>This deployment flow simplifies the steps below and you will only need to know these steps if you intend on using the operator console to operate a Blockchain network. If you do wish to operate a network in the future, or just want to know about how the smart contract deployment works in Fabric v2.0, then continue reading.</p>
                    </div>
                    <div className='bx--row margin-bottom-09'>
                        <div className='bx--col margin-right-07 padding-right-05'>
                            <p className='step-title margin-bottom-05'>Step 1</p>
                            <h6 className='margin-bottom-05'>Your smart contract is packaged</h6>
                            <p className='margin-bottom-05'>We need to package the chaincode before it can be installed on our peers.</p>
                        </div>
                        <div className='bx--col'>
                            <ThemedImage darkImg={darkStep1} lightImg={lightStep1} altText='' id='step-1-deploy' className=''/>
                        </div>
                    </div>
                    <div className='bx--row margin-bottom-09'>
                        <div className='bx--col'>
                            <ThemedImage darkImg={darkStep2} lightImg={lightStep2} altText='' id='step-2-deploy'/>
                        </div>
                        <div className='bx--col margin-left-07 padding-left-05'>
                            <p className='step-title margin-bottom-05'>Step 2</p>
                            <h6 className='margin-bottom-05'>Each of the network's member organizations install the package on their peers</h6>
                            <p className='margin-bottom-05'>After we package the smart contract, we can install the chaincode on our peers. The chaincode needs to be installed on every peer that will endorse a transaction.</p>
                        </div>
                    </div>
                    <div className='bx--row margin-bottom-09'>
                        <div className='bx--col margin-right-07 padding-right-05'>
                            <div className='title-and-tag margin-bottom-05'>
                                <span>Step 3</span>
                                <Tag type='blue' className='fabric-2-tag'>New for Fabric v2.0</Tag>
                            </div>
                            <h6 className='margin-bottom-05'>Each organization approves a shared definition of what they will use on the channel</h6>
                            <p className='margin-bottom-05'>After you install the chaincode package, you need to approve a chaincode definition for your organisation. The definition includes important parameters of chaincode governance such as the name, version, and the chaincode endorsement policy.</p>
                        </div>
                        <div className='bx--col'>
                            <ThemedImage darkImg={darkStep3} lightImg={lightStep3} altText='' id='step-3-deploy'/>
                        </div>
                    </div>
                    <div className='bx--row margin-bottom-09'>
                        <div className='bx--col'>
                            <ThemedImage darkImg={darkStep4} lightImg={lightStep4} altText='' id='step-4-deploy'/>
                        </div>
                        <div className='bx--col margin-left-07 padding-left-05'>
                            <div className='title-and-tag margin-bottom-05'>
                                <span>Step 4</span>
                                <Tag type='blue' className='fabric-2-tag'>New for Fabric v2.0</Tag>
                            </div>
                            <h6 className='margin-bottom-05'>The definition is then committed: a transaction endorsed by the channel members</h6>
                            <p className='margin-bottom-05'>After a sufficient number of organizations have approved a chaincode definition, one organization can commit the chaincode definition to the channel. If a majority of channel members have approved the definition, the commit transaction will be successful and the parameters agreed to in the chaincode definition will be implemented on the channel.</p>
                        </div>
                    </div>
                    <div className='bx--row'>
                        <p>To learn more, visit the <Link href='https://hyperledger-fabric.readthedocs.io/en/release-2.0/chaincode_lifecycle.html' id='deploy-step-1-link'>Fabric chaincode lifecycle</Link> section in the Hyperledger Fabric docs.</p>
                    </div>
                </AccordionItem>
            </Accordion>
        );

    } else {
        explanationJSX = (
            <p>To find out more about Fabric v1.4 smart contract deployment, check out the documentation available <Link href='https://hyperledger-fabric.readthedocs.io/en/release-1.4/chaincode4noah.html' id='deploy-step-1-link'>here</Link>.</p>
        );
    }

    return explanationJSX;
};

export default DeployExplanation;
