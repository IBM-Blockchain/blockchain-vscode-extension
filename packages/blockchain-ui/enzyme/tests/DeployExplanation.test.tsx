import React from 'react';
import renderer from 'react-test-renderer';
import chai from 'chai';
import sinonChai from 'sinon-chai';
import DeployExplanation from '../../src/components/elements/DeployExplanation/DeployExplanation';
import TransactionOutput from '../../src/components/elements/TransactionOutput/TransactionOutput';
chai.should();
chai.use(sinonChai);

describe('DeployExplanation component', () => {
    it('should render the expected snapshot when showing the v2 deploy view', () => {
        const component: any = renderer
            .create(<DeployExplanation showV2Explanation={true}/>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should render the expected snapshot when showing the v1 deploy view', () => {
        const component: any = renderer
            .create(<DeployExplanation showV2Explanation={false}/>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
});
