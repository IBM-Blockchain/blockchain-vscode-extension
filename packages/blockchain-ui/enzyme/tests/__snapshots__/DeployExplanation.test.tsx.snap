// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeployExplanation component should render the expected snapshot when showing the v1 deploy view 1`] = `
<p>
  To find out more about Fabric v1.4 smart contract deployment, check out the documentation available 
  <a
    className="bx--link"
    href="https://hyperledger-fabric.readthedocs.io/en/release-1.4/chaincode4noah.html"
    id="deploy-step-1-link"
  >
    here
  </a>
  .
</p>
`;

exports[`DeployExplanation component should render the expected snapshot when showing the v2 deploy view 1`] = `
<ul
  className="bx--accordion bx--accordion--end"
>
  <li
    className="bx--accordion__item"
    onAnimationEnd={[Function]}
  >
    <button
      aria-expanded={false}
      className="bx--accordion__heading"
      onClick={[Function]}
      onKeyDown={[Function]}
      title="Expand/Collapse"
      type="button"
    >
      <svg
        aria-label="Expand/Collapse"
        className="bx--accordion__arrow"
        focusable="false"
        height={16}
        preserveAspectRatio="xMidYMid meet"
        role="img"
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          points="11,8 6,13 5.3,12.3 9.6,8 5.3,3.7 6,3"
        />
      </svg>
      <div
        className="bx--accordion__title"
      >
        How does Fabric v2.X smart contract deployment work?
      </div>
    </button>
    <div
      className="bx--accordion__content"
    >
      <div
        className="bx--row margin-top-05 margin-bottom-09"
      >
        <p>
          This deployment flow simplifies the steps below and you will only need to know these steps if you intend on using the operator console to operate a Blockchain network. If you do wish to operate a network in the future, or just want to know about how the smart contract deployment works in Fabric v2.0, then continue reading.
        </p>
      </div>
      <div
        className="bx--row margin-bottom-09"
      >
        <div
          className="bx--col margin-right-07 padding-right-05"
        >
          <p
            className="step-title margin-bottom-05"
          >
            Step 1
          </p>
          <h6
            className="margin-bottom-05"
          >
            Your smart contract is packaged
          </h6>
          <p
            className="margin-bottom-05"
          >
            We need to package the chaincode before it can be installed on our peers.
          </p>
        </div>
        <div
          className="bx--col"
        >
          <div
            id="step-1-deploy-themed-image"
          >
            <img
              alt=""
              className=""
              src="test-file-stub"
            />
            <img
              alt=""
              className=""
              src="test-file-stub"
            />
          </div>
        </div>
      </div>
      <div
        className="bx--row margin-bottom-09"
      >
        <div
          className="bx--col"
        >
          <div
            id="step-2-deploy-themed-image"
          >
            <img
              alt=""
              className=""
              src="test-file-stub"
            />
            <img
              alt=""
              className=""
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          className="bx--col margin-left-07 padding-left-05"
        >
          <p
            className="step-title margin-bottom-05"
          >
            Step 2
          </p>
          <h6
            className="margin-bottom-05"
          >
            Each of the network's member organizations install the package on their peers
          </h6>
          <p
            className="margin-bottom-05"
          >
            After we package the smart contract, we can install the chaincode on our peers. The chaincode needs to be installed on every peer that will endorse a transaction.
          </p>
        </div>
      </div>
      <div
        className="bx--row margin-bottom-09"
      >
        <div
          className="bx--col margin-right-07 padding-right-05"
        >
          <div
            className="title-and-tag margin-bottom-05"
          >
            <span>
              Step 3
            </span>
            <span
              className="bx--tag bx--tag--blue fabric-2-tag"
            >
              New for Fabric v2.0
            </span>
          </div>
          <h6
            className="margin-bottom-05"
          >
            Each organization approves a shared definition of what they will use on the channel
          </h6>
          <p
            className="margin-bottom-05"
          >
            After you install the chaincode package, you need to approve a chaincode definition for your organisation. The definition includes important parameters of chaincode governance such as the name, version, and the chaincode endorsement policy.
          </p>
        </div>
        <div
          className="bx--col"
        >
          <div
            id="step-3-deploy-themed-image"
          >
            <img
              alt=""
              className=""
              src="test-file-stub"
            />
            <img
              alt=""
              className=""
              src="test-file-stub"
            />
          </div>
        </div>
      </div>
      <div
        className="bx--row margin-bottom-09"
      >
        <div
          className="bx--col"
        >
          <div
            id="step-4-deploy-themed-image"
          >
            <img
              alt=""
              className=""
              src="test-file-stub"
            />
            <img
              alt=""
              className=""
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          className="bx--col margin-left-07 padding-left-05"
        >
          <div
            className="title-and-tag margin-bottom-05"
          >
            <span>
              Step 4
            </span>
            <span
              className="bx--tag bx--tag--blue fabric-2-tag"
            >
              New for Fabric v2.0
            </span>
          </div>
          <h6
            className="margin-bottom-05"
          >
            The definition is then committed: a transaction endorsed by the channel members
          </h6>
          <p
            className="margin-bottom-05"
          >
            After a sufficient number of organizations have approved a chaincode definition, one organization can commit the chaincode definition to the channel. If a majority of channel members have approved the definition, the commit transaction will be successful and the parameters agreed to in the chaincode definition will be implemented on the channel.
          </p>
        </div>
      </div>
      <div
        className="bx--row"
      >
        <p>
          To learn more, visit the 
          <a
            className="bx--link"
            href="https://hyperledger-fabric.readthedocs.io/en/release-2.0/chaincode_lifecycle.html"
            id="deploy-step-1-link"
          >
            Fabric chaincode lifecycle
          </a>
           section in the Hyperledger Fabric docs.
        </p>
      </div>
    </div>
  </li>
</ul>
`;
